File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    ans = 1
    # 2번 3번이면 바로 종료하네
    
    
    
    while abs((a // 2 + a % 2) - (b // 2 + b % 2)) != 0:
        a = a // 2 + a % 2
        b = b // 2 + b % 2
        ans += 1
    return ans

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 1;
        while(n > 2) {
            a = (a + 1) / 2;
            b = (b + 1) / 2;
            if(a == b) {
                return answer;
            }
            answer++;
            n /= 2;
        }
        return answer;
    }
}

/*
1 2 3 4 5 6 7 8
 1   2   3   4
   1       2
       1

*/